models/analytics/intermediate/d_shifts.sql

```diff
@@ -1,25 +1,164 @@
+            when schedule.game_type = '02' and shifts.period = 5 then 'shootout'
+            when shifts.period > 3 then 'overtime'
+            when shifts.period between 1 and 3 then 'regulation'
+            else 'unknown'
+        end as period_type
+
+        /* Flags */
+        , case
+            when schedule.game_type = '02' and shifts.period = 5 then false
+            when shifts.type_code = 505 then true
+            else false
+        end as is_goal
+        , shifts.is_period_start
+        , shifts.is_period_end
+    from {{ ref('stg_nhl__shifts') }} as shifts
+    left join {{ ref('d_schedule') }} as schedule
+        on shifts.game_id = schedule.game_id
+    where 1 = 1
+        and not (schedule.game_type = '02' and shifts.period = 5) -- remove shootouts
+        and shifts.end_time != '' -- remove 1129 shifts, dups
+)
+
+
+, find_duplicate_shifts as (
+    -- as of 7/10/2022, there were 91 shifts that mapped to 13 plays where the shift was duplicated having the same start time but different durations (thus, different end times)
+    -- rule: if a player's shift is duplicated, take the shift that was longer since they might have been involved in the play (min takes shorter duration, max takes longer)
+    select
+        shift_id
+        , game_id
+        , player_id
+        , period
+        , start_seconds_elapsed
+        , event_number
+        , shift_number
+        , row_number() over(
+            partition by
+                game_id
+                , player_id
+                , period
+                , start_seconds_elapsed
+            order by
+                period asc
+                , start_seconds_elapsed asc
+                , duration desc
+        ) as duplicate_sequence_number -- 1 = shift among duplicates with the longest duration, >1 means duplicates we don't care about
+    from shifts
+    where not is_goal
+)
+
+, deduped_shift_attributes as (
+    -- return the original shift and event numbers that were identified as deduplicates as an array
+    select
+        game_id
+        , player_id
+        , period
+        , start_seconds_elapsed
+        , string_agg(cast(shift_number as string)) as shift_numbers
+        , string_agg(cast(event_number as string)) as event_numbers
+        , string_agg(cast(shift_id as string)) as shift_ids
+    from find_duplicate_shifts
+    group by 1, 2, 3, 4
+)
+
+, revised_shift_number_for_duplicate_shifts as (
+    -- create the new_shift_number, excluding any shifts that were duplicated as well as goals
+    select
+        shift_id
+        , game_id
+        , player_id
+        , start_seconds_elapsed
+        , period
+        , shift_number as original_shift_number
+        , row_number() over(
+            partition by
+                game_id
+                , player_id
+            order by
+                start_seconds_elapsed asc
+        ) as revised_shift_number
```

Comment:
everything checks out, however when I compared this to the previous version, I see slightly more shifts than before... strangely enough, only for the last 2 seasons..

```
select season_id, count(1) as ct, count(distinct b.shift_id) as shifts, 'gavin' as type
from `nhl-breakouts.dbt_gavin.d_shifts` as b
left join `nhl-breakouts.dbt_gavin.d_schedule` as s on s.game_id = b.game_id
group by 1
union all
select season_id, count(1) as ct, count(distinct b.shift_id) as shifts, 'dom' as type
from `nhl-breakouts.dbt_dom.stg_nhl__shifts` as b
left join `nhl-breakouts.dbt_dom.d_schedule` as s on s.game_id = b.game_id
group by 1 order by 1
;
```

If you run this, you will see what I mean... `shift_id` should be 1:1, however, for 2020-21 & 2021-22, they are not... very small diff, but may be worth looking into... 
![image](https://user-images.githubusercontent.com/48701668/204073485-a4aa5094-6935-42f7-a3eb-7e66d982b9c4.png)